app/scripts/modules/amazon/src/function/details/FunctionActions.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { Dropdown } from 'react-bootstrap';
+
+import { Application, ApplicationReader, FunctionWriter, SETTINGS, NgReact, ReactInjector } from '@spinnaker/core';
+
+import { IAmazonFunction, IAmazonFunctionDeleteCommand } from 'amazon/domain';
+
+import { IFunctionFromStateParams } from './AmazonFunctionDetails';
+import { CreateLambdaFunction } from '../CreateLambdaFunction';
+
+export interface IFunctionActionsProps {
+  app: Application;
+  functionDef: IAmazonFunction;
+  functionFromParams: IFunctionFromStateParams;
+}
+
+export interface IFunctionActionsState {
+  application: Application;
+}
+
+export class FunctionActions extends React.Component<IFunctionActionsProps, IFunctionActionsState> {
+  constructor(props: IFunctionActionsProps) {
+    super(props);
+
+    const { app, functionDef } = this.props;
+    let application: Application;
+
+    const functionAppName = functionDef.functionName.split('-')[0];
+    if (functionAppName === app.name) {
+      // Name matches the currently active application
+      application = app;
+    } else {
+      // Load balancer is a part of a different application
+      ApplicationReader.getApplication(functionAppName)
+        .then(functionApp => {
+          this.setState({ application: functionApp });
+        })
+        .catch(() => {
+          // If the application can't be found, just use the old one
+          this.setState({ application: this.props.app });
+        });
+    }
```

Comment:
Typically the constructor is only used for assigning initial state, and not using `setState`. Since you have some async behavior, then I would recommend moving this logic to `componentDidMount` instead.

Reply:
I moved it to componentWillMount.

Will appear in the next push.

Reply:
`componentWillMount` is deprecated, I also recommend `componentDidMount`

Reply:
`render()` is called before `componentDidMount()` and `application` variable is needed in `render()`.
That is why this `setState` was in the constructor.

Reply:
You can set a default value in the constructor (`this.props.app` seems like a viable option), then use `componentDidMount` to immediately fetch the application and update the state if needed. 

If you need the values on render then you can also consider defaultng `application` to `null` and then render a loading spinner until the fetch has completed.